xbmc/interfaces/json-rpc/PlayerOperations.cpp

```diff
@@ -163,12 +163,12 @@ JSONRPC_STATUS CPlayerOperations::GetItem(const std::string &method, ITransportL
             if (fileItem->GetLabel().empty())
               fileItem->SetLabel(originalLabel);
           }
-          fileItem->SetPath(g_application.CurrentFileItem().GetPath());
+          fileItem->SetPath(fileItem->GetPath());
```

Comment:
@FernetMenta this line is what looks most odd to me now. I want to rethink it before merge

Reply:
why does it look odd to you?

Reply:
right :) basically it is the same code as before but now it is more obvious that this line was nonsense.

Reply:
I was wrong, it is not the same behaviour as before but I did not get the sense of what it was intended to do

```
          if (!CAudioLibrary::FillFileItem(g_application.CurrentFile(), fileItem, parameterObject))
          {
            const MUSIC_INFO::CMusicInfoTag *currentMusicTag = g_infoManager.GetCurrentSongTag();
            if (currentMusicTag != NULL)
              fileItem = CFileItemPtr(new CFileItem(*currentMusicTag));
            fileItem->SetPath(g_application.CurrentFileItem().GetPath());
          }
```

it changes file item to represent video tag, then set original path. all this strange logic is misplaced in what is supposed to be an API.

Reply:
I looked back through Git to try and understand why @Montellese added it. I believe he had a coherent design in mind  even if it is not obvious now to us what that was. The idea was to use the item data in `CGUIInfoManager` current video or music tag as a fallback if looking it up for the app current item path was unsuccessfull, without changing the path. See https://github.com/xbmc/xbmc/pull/786/commits/73fd7b21ce4bc5d53c7e37e497a64132648b3110. It was subsequently amended to allow for a null tag pointers etc., but that was the introduction back in 2012.

At that time the app current item was not the definative data, CGUIInfoManager was taken as a valid source. I also think that addons can edit the GUI item data, and since the API is trying to provide consumers with whatever data the GUI has (be that from db or elsewhere), then the processing here looks less strange. 

With the changes in v18 to `CApplication::m_itemCurrentFile` do we still need to look for fallback item data in the GUI item? I'm not sure. The fallback comes into effect when playing non-library items, can GUI current item have more details about the item than the app current item? I think it might, so at this point I think we are best keeping this fallback, but preserving the initial path without accessing `CApplication::m_itemCurrentFile`  a second time in case it has already changed on the main thread.

As for being misplaced, well yes it would it be better if some other common something was providing both GUI, Python API and JSON API with consistent full item data, but fixing a seg fault in beta I don't think we will get there today.

Reply:
CApplication::m_itemCurrentFile is single source of truth. Changing db entry or guiInfo must only be done by Capp.
Hence any manipulation of fileItem here is wrong and should go away. This is true regardless of the item is in library or not.
Accessing guiInfo is even worse than accessing db because json-rpc is supposed to work when kodi runs headless.

In the current state the addition code does not really harm because, provided all goes right, it updates fimeItem with its own values.
Please place a todo comment that all this code should be dropped in v19.

Please also drop the api bump because this is not an api change.

Reply:
>CApplication::m_itemCurrentFile is single source of truth.
Yes @FernetMenta I know that is what is wanted, but it has not been achieved in v18. The code here does not manipulate anything, it simply continues to accommodate the fact that CApplication::m_itemCurrentFile does not always have all the item details that GUI item does.

>Please place a todo comment that all this code should be dropped in v19.
I have added @todo, but it is not just about dropping this code that would simply cause a regression. More changes need to be made elsewhere to achieve `CApplication::m_itemCurrentFile`   as the  single source of truth.